index.xml

```diff
@@ -14109,6 +14240,155 @@ C implements A;</x:codeblock>
           </ol>
         </div>
 
+        <div id='es-namespaces' class='section'>
+          <h3>Namespaces</h3>
+
+          <p>
+            For every <a class='dfnref' href='#dfn-namespace'>namespace</a> that is <a
+            class='dfnref' href='#dfn-exposed'>exposed</a> in a given ECMAScript global environment,
+            a corresponding property <span class='rfc2119'>MUST</span> exist on the ECMAScript
+            environment's global object. The name of the property is the <a class='dfnref'
+            href='#dfn-identifier'>identifier</a> of the namespace, and its value is an object
+            called the <dfn data-export='' id='dfn-namespace-object'>namespace object</dfn>.
+          </p>
+          <p>
+            The property has the attributes <span class='descriptor'>{ [[Writable]]: <span
+            class='esvalue'>true</span>, [[Enumerable]]: <span class='esvalue'>false</span>,
+            [[Configurable]]: <span class='esvalue'>true</span> }</span>. The characteristics of a
+            namespace object are described in <a href='#namespace-object'>section <?sref
+            namespace-object?></a> <?sdir namespace-object?>.
+          </p>
+
+          <div id='namespace-object' class='section'>
+            <h4>Namespace object</h4>
+
+            <p>
+              The namespace object for a given <a class='dfnref' href='#dfn-namespace'>namespace</a>
+              <var>namespace</var> and Realm <var>realm</var> is created as follows:
+            </p>
+
+            <ol class='algorithm'>
+              <li>
+                Let <var>namespaceObject</var> be
+                ! <a>ObjectCreate</a>(the <a>%ObjectPrototype%</a> of <var>realm</var>).
```

Comment:
I think this syntax is terrible, as I've pointed out before.  The fact that it's (1) non-obvious and (2) not linkified to point to its definition makes it _really_ hard to get into reading a spec without reading a bunch of front matter that hides the syntax definition quite well.

But also, nothing in Web IDL links to that section of the ES spec, so people would have no idea you're doing that anyway.  This would be a bare minimum requirement for using this syntax, because then you could at least make the (laughable, imo, but technically correct) claim that "if you had read the entire Web IDL spec starting at the very beginning and to this point, and followed all the links from it and read those too, you would know what this syntax means in this spec".  A slight step up would be explicitly saying that '!' and '?" mean what they mean in `#sec-algorithm-conventions`, or that all algorithms are written using those conventions (and hence the reader of Web IDL should read them) or something.